File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;   
    queue<int> q;
    
    for(int i = 0;i < progresses.size();i++){
        int a = (100-progresses[i])/speeds[i];
        int b = (100-progresses[i])%speeds[i];
        if (b!=0) a += 1;
        q.push(a);
    }
    
    while(!q.empty()){
        int c = q.front();
        q.pop();
        int count = 1;
        
        while(!q.empty() && q.front() <= c){
                count++;
                q.pop();
        }
        answer.push_back(count); 
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> result = new ArrayList<>();
        Queue<Integer> q = new LinkedList<>();
        
        for (int i = 0; i < progresses.length; i++) {
            int remainingWork = 100 - progresses[i];
            int days = (int)Math.ceil((double)remainingWork / speeds[i]);  
            q.offer(days);
        }
        
        while (!q.isEmpty()) {
            int currentDay = q.poll();
            int count = 1;
            
            while (!q.isEmpty() && q.peek() <= currentDay) {
                q.poll();
                count++;
            }
            
            result.add(count);
        }
        
        return result.stream().mapToInt(i -> i).toArray();
    }
}
